static/extensions/MubiLop/penguingpt.js

```diff
@@ -0,0 +1,752 @@
+                        prompt: prompt
+                    }),
+                })
+                .then(response => {
+                    if (!response.ok) {
+                        throw new Error(`Network response was not ok: ${response.status} ${response.statusText}`);
+                    }
+                    return response.json();
+                })
+                .then(data => {
+                    let botResponse
+                    if (requestedModel === "dalle-3") {
+                        botResponse = JSON.stringify(data);
+                    } else {
+                        botResponse = data.results
+                    }
+                    return botResponse;
+                })
+                .catch(error => {
+                    console.error("Error sending prompt to Image Generator", error.message);
+                    return "Error: ", error.message;
+                });
+        }
+        generateImageAndImport(args) {
+            const prompt = args.PROMPT;
+            const requestedModel = args.MODEL
+            const Name = args.NAME || `AIGenerated_${prompt}`;
+
+            return Scratch.fetch("https://reverse.mubi.tech/image/generate", {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json'
+                    },
+                    body: JSON.stringify({
+                        model: requestedModel,
+                        prompt: prompt
+                    }),
+                })
+                .then(response => {
+                    if (!response.ok) {
+                        throw new Error(`Network response was not ok: ${response.status} ${response.statusText}`);
+                    }
+                    return response.json();
+                })
+                .then(data => {
+                    let botResponse
+                    if (requestedModel === "dalle-3") {
+                        fetch(data.url)
+                        .then((r) => r.arrayBuffer())
+                        .then((arrayBuffer) => {
+                            const storage = vm.runtime.storage;
+                            vm.addCostume(Name + '.PNG', {
+                            name: Name,
+                            asset: new storage.Asset(
+                                storage.AssetType.ImageBitmap,
+                                null,
+                                storage.DataFormat.PNG,
+                                new Uint8Array(arrayBuffer),
+                                true
+                            )
+                            })
+                        });
+                    } else {
+                        fetch(data.results)
+                        .then((r) => r.arrayBuffer())
+                        .then((arrayBuffer) => {
+                            const storage = vm.runtime.storage;
+                            vm.addCostume(Name + '.PNG', {
```

Comment:
target id is not provided so this will add to the editor's editingTarget, instead of adding the costume to the sprite the block was used in

Reply:
Well how can I fix it?